cmd/otel-allocator/README.md

```diff
@@ -4,6 +4,135 @@ The TargetAllocator is an optional separately deployed component of an OpenTelem
+
+A basic example that deploys, but does nothing.
+```yaml
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: collector-with-ta
+spec:
+  mode: statefulset
+  targetAllocator:
+    enabled: true
+  config: |
+    receivers:
+      prometheus:
+        config:
+          scrape_configs:
+          - job_name: 'otel-collector'
+
+    exporters:
+      logging:
+
+    service:
+      pipelines:
+        traces:
+          receivers: [prometheus]
+          processors: []
+          exporters: [logging]
+```
+
+## PrometheusCR specifics
+TargetAllocator discovery of PrometheusCRs can be turned on with
+```yaml
+  targetAllocator:
+    enabled: true
+    prometheusCR:
+      enabled: true
+```
+The CRs can be filtered by labels as documented here: [api.md#opentelemetrycollectorspectargetallocatorprometheuscr](../../docs/api.md#opentelemetrycollectorspectargetallocatorprometheuscr)
+
+The prometheus receiver in the deployed collector also has to know where the Allocator service exists. This is done by a
+OpenTelemetry Collector operator specific config.
+```yaml
+  config: |
+    receivers:
+      prometheus:
+        config:
+          scrape_configs:
+          - job_name: 'otel-collector'
+        target_allocator:
+          endpoint: http://my-targetallocator-service
+          interval: 30s
+          collector_id: collector-1
+```
+Upstream documentation here: [Prometheusreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver#opentelemetry-operator)
+
+The TargetAllocator service is named based on the OpenTelemetryCollector CR name. `collector_id` should be unique per
+collector instance, such as the pod name. This can be passed down to the pod with env vars and replaced in the config
+with `${ENV}` notation.
+
+The CRDs naturally also has to exist for the Allocator to pick them up. The best place to get them is from
+prometheus-operator: [Releases](https://github.com/prometheus-operator/prometheus-operator/releases). Only the CRDs for
+CRs that the Allocator watches for need to be deployed. They can be picked out from the bundle.yaml file.
+
+### RBAC
+The ServiceAccount that the TargetAllocator runs as, has to have access to the CRs. A role like this will provide that
+access.
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: opentelemetry-targetallocator-cr-role
+rules:
+- apiGroups:
+  - monitoring.coreos.com
+  resources:
+  - servicemonitors
+  - podmonitors
+  verbs:
+  - '*'
+```
+In addition, the TargetAllocator needs the same permissions as a Prometheus instance would to find the matching targets
+from the CR instances.
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: opentelemetry-targetallocator-role
+rules:
+- apiGroups: [""]
+  resources:
+  - nodes
+  - nodes/metrics
+  - services
+  - endpoints
+  - pods
```

Comment:
Unsure if we actually need all of these

Reply:
i think all of these are required if someone wants to use the `kubernetes_sd_config`